addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/TestRetentionManager.java

```diff
@@ -0,0 +1,218 @@
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.Arrays;
+import java.util.Calendar;
+
+import javax.inject.Inject;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.junit.Test;
+import org.nuxeo.ecm.automation.AutomationService;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.api.event.DocumentEventTypes;
+import org.nuxeo.ecm.core.bulk.BulkService;
+import org.nuxeo.retention.adapters.Record;
+import org.nuxeo.retention.adapters.RetentionRule;
+
+/**
+ * @since 11.1
+ */
+public class TestRetentionManager extends RetentionTestCase {
+
+    public static Log log = LogFactory.getLog(TestRetentionManager.class);
+
+    @Inject
+    protected AutomationService automationService;
+
+    @Inject
+    protected BulkService bulkService;
+
+    @Test
+    public void testRuleOnlyFile() throws InterruptedException {
+        DocumentModel workspace = session.createDocumentModel("/", "workspace", "Workspace");
+        workspace = session.createDocument(workspace);
+        workspace = session.saveDocument(workspace);
+        try {
+            service.attachRule(workspace, createManualImmediateRuleMillis(100L), session);
+            fail("Should not accept workspace document");
+        } catch (NuxeoException e) {
+            // expected
+        }
+    }
+
+    @Test
+    public void test1DayManualImmediateRuleRunningRetention() throws InterruptedException {
+        assertStillUnderRetentionAfter(file, createRuleWithActions(RetentionRule.ApplicationPolicy.MANUAL,
+                RetentionRule.StartingPointPolicy.IMMEDIATE, null, null, null, null, 0L, 0L, 1L, 0L, null, null), 1_000);
+    }
+
+    @Test
+    public void test1MonthManualImmediateRuleRunningRetention() throws InterruptedException {
+        assertStillUnderRetentionAfter(file, createRuleWithActions(RetentionRule.ApplicationPolicy.MANUAL,
+                RetentionRule.StartingPointPolicy.IMMEDIATE, null, null, null, null, 0L, 1L, 0L, 0L, null, null), 1_000);
+    }
+
+    @Test
+    public void test1YearManualImmediateRuleRunningRetention() throws InterruptedException {
+        assertStillUnderRetentionAfter(file, createRuleWithActions(RetentionRule.ApplicationPolicy.MANUAL,
+                RetentionRule.StartingPointPolicy.IMMEDIATE, null, null, null, null, 1L, 0L, 0L, 0L, null, null), 1_000);
+    }
+
+    @Test
+    public void testManualImmediateRuleWithActions() throws InterruptedException {
+        RetentionRule testRule = createImmediateRuleMillis(RetentionRule.ApplicationPolicy.MANUAL, 100L,
+                null, Arrays.asList("Document.Trash"));
+
+        file = service.attachRule(file, testRule, session);
+        assertTrue(session.isRecord(file.getRef()));
+        assertTrue(session.isUnderRetentionOrLegalHold(file.getRef()));
+
+        awaitRetentionExpiration(1000L);
```

Comment:
can be replaced by `1_000` and i think the `L` litteral is not needed here (awaitRetentionExpiration take a primitif type), i used `awaitRetentionExpiration(1_000);` and the code compile

Reply:
Removing useless `L`, but `1000` is readable enough as it is.